graph.js

```diff
@@ -157,4 +180,95 @@ function drawGraph(graph) {
     d.fy = null;
   }
 
+  function addLegend() {
+    var legendData = d3.scaleOrdinal()
+      .domain(["GOOD - connected to Command N Control (CNC) node", "BAD - bad response", "MISSING - not connected to CNC but known to peers via libp2p"])
+      .range([ '#1f77b4', '#aec7e8', '#ff7f0e' ]);
+
+    var svg = d3.select("svg");
+
+    var legend = svg.append("g")
+      .attr("class", "legend")
+      .attr("transform", "translate(20,20)")
+
+    var legendRect = legend
+      .selectAll('g')
+      .data(legendData.domain());
+
+    var legendRectE = legendRect.enter()
+      .append("g")
+      .attr("transform", function(d,i){
+        return 'translate(0, ' + (i * 20) + ')';
+      });
+
+    legendRectE
+      .append('path')
+      .attr("d", d3.symbol().type(d3.symbolCircle))
+      .style("fill", function (d,i) {
+          return legendData(i);
+      });
+
+    legendRectE
+      .append("text")
+      .attr("x", 10)
+      .attr("y", 5)
+      .text(function (d) {
+          return d;
+      });
+
+  } // end addLegend()
+
+  // Inline d3 legend extension (require wasn't working right)
+  // d3.legend.js
+  // (C) 2012 ziggy.jonsson.nyc@gmail.com
+  // MIT licence
+  d3.legend = function(g) {
```

Comment:
are we using this? i dont see it called anywhere

Reply:
No, you're right its old! Sorry. It was originally grabbing attribute data from nodes - but could be used for counts. Will remove in an update.